### PR TITLE
[POAE7-2333] Arrow based batch test framework

### PR DESCRIPTION
**Pull request by jikunshang**
### What changes were proposed in this pull request?

1. reuse current cider batch data generator, and convert to arrow based batch in test. will not convert null data currently.
2. for result check, just compare row/column num with duck result, didn't verify actual data.
3. will replace with new utils.

### CODE REVIEW

**Comment by jikunshang** [comment]
@bigPYJ1151 @xieqi @yma11 Please take a review.

File: cider/exec/module/batch/CiderBatch.cpp [hunk 1]
```diff
@@ -253,6 +253,51 @@ void CiderBatch::releaseArrowEntries() {
   }
 }
 
+// to be deprecated, just for test not nullable data.
+void CiderBatch::convertToArrowRepresentation() {
```
Comment (spevenhe) [inline cider/exec/module/batch/CiderBatch.cpp:257]: This is converting to structbatch right? if column_num() is 1, should ciderbatch convert to strcutbatch or scalarbatch?

Comment (jikunshang) [reply cider/exec/module/batch/CiderBatch.cpp:257]: yes. I think even a single column batch should convert to structBatch.

Comment (spevenhe) [reply cider/exec/module/batch/CiderBatch.cpp:257]: Then what is the use of scalarbatch?

Comment (jikunshang) [reply cider/exec/module/batch/CiderBatch.cpp:257]: I think codegen internal will never use scalarBatch. maybe it's just for test. Correct me if I am wrong. @bigPYJ1151

File: cider/tests/utils/CiderTestBase.cpp [hunk 2]
```diff
@@ -41,6 +41,23 @@ void CiderTestBase::assertQuery(const std::string& sql,
...
 
+void CiderTestBase::prepareArrowBatch() {
+  for (auto iter = input_.begin(); iter != input_.end(); iter++) {
+    (*iter)->convertToArrowRepresentation();
+  }
+}
+
+void CiderTestBase::assertQueryArrow(const std::string& sql,
+                                     const std::string& json_file) {
+  auto duck_res = duckDbQueryRunner_.runSql(sql);
```
Comment (yma11) [inline cider/tests/utils/CiderTestBase.cpp:52]: currently duckdb input is still converted based on original format, so we also need to update the convertor based on arrow format right?

Comment (jikunshang) [reply cider/tests/utils/CiderTestBase.cpp:52]: yes, borui is working on this.

File: cider/exec/template/NativeCodegen.cpp [hunk 3]
```diff
@@ -1896,8 +1896,9 @@ bool Executor::compileBody(const RelAlgExecutionUnit& ra_exe_unit,
       filter_lv = cgen_state_->ir_builder_.CreateAnd(
           filter_lv, cgen_state_->ir_builder_.CreateNot(is_null));
     }
-    return cgen_state_->ir_builder_.CreateAnd(filter_lv,
-                                              fixedsize_expr_value->getValue());
+    // need to convert to bool expr first.
+    auto cond = code_generator.toBool(fixedsize_expr_value->getValue());
```
Comment (bigPYJ1151) [inline cider/exec/template/NativeCodegen.cpp:1900]: Confused, results of filter related expressions (logicalOp, cmpOP, .etc) should be bool type in my mind. Any special cases here?

Comment (jikunshang) [reply cider/exec/template/NativeCodegen.cpp:1900]: constant need this. query like `SELECT col_1 FROM test WHERE 2 = 2`.

**Review by yma11 — APPROVED** [review]
I am okay to merge this so that we can start using the CiderBatchToArrow util for more migration.

**Review by bigPYJ1151 — APPROVED** [review]

**Pull request merged**